.azure-pipelines/collect_dump.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import sys
+import datetime
+import traceback
+import tarfile
+import gzip
+
+_self_dir = os.path.dirname(os.path.abspath(__file__))
+base_path = os.path.realpath(os.path.join(_self_dir, ".."))
+if base_path not in sys.path:
+    sys.path.append(base_path)
+ansible_path = os.path.realpath(os.path.join(_self_dir, "../ansible"))
+if ansible_path not in sys.path:
+    sys.path.append(ansible_path)
+
+from devutil.devices.factory import init_testbed_sonichosts  # noqa: E402
+
+logger = logging.getLogger(__name__)
+
+RC_INIT_FAILED = 1
+RC_GET_TECHSUPPORT_FAILED = 2
+TECHSUPPORT_SAVE_PATH = '../tests/logs/'
+LOGS_DIR = os.path.join(_self_dir, TECHSUPPORT_SAVE_PATH)
+
+
+def get_techsupport(dut, time_since):
```

Comment:
Why not pass in `sonichosts` and run `sonichosts.command(f"show techsupport --since {time_since}")`. This would take advantage of the lib's forking capability to run in parallel on multiple sonichosts.

The `show techsupport` command takes time. Run in parallel can save execution time.

Reply:
fixed